source/math/FloatTransform3D.ooc

```diff
@@ -208,7 +208,7 @@ FloatTransform3D: cover {
 	}
 	transformAndProject: func ~FloatPoint2D (point: FloatPoint2D, focalLength: Float) -> FloatPoint2D {
 		transformedWorldPoint := this * FloatPoint3D new(point x, point y, focalLength)
-		this project(transformedWorldPoint, focalLength)
+		focalLength == 0.0f ? FloatPoint2D new(transformedWorldPoint x, transformedWorldPoint y) : this project(transformedWorldPoint, focalLength)
```

Comment:
https://en.wikipedia.org/wiki/Focal_length
As I understand about working camera lens, focal length should be a positive number (greater than 0).

You can read link below to know when focal length is negative.
https://www.rp-photonics.com/focal_length.html